docs/proposals/api-machinery/20230707-add-compatibility-interface-for-priority-class.md

```diff
@@ -0,0 +1,105 @@
+
+## Table of Contents
+
+- [Add compatibility interface for priority class](#add-compatibility-interface-for-priority-class)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+    - [Implementation](#implementation)
+  - [Alternatives](#alternatives)
+  - [Unsolved Problems](#unsolved-problems)
+  - [Implementation History](#implementation-history)
+
+## Summary
+
+Provide a solution using Koordinator PriorityClass for already running a large number of Pod clusters by adding a label `koordinator.sh/priority-class`.
+
+## Motivation
+
+Koordinator defines a set of specifications on top of kubernetes priority class, and extends a dimension of to support fine-grained colocation(see [Koordinator Priority Class](https://koordinator.sh/docs/architecture/priority)). 
+If your cluster follows Koordinator's best practice to build the entire scheduling system from initialization, then everything will be very natural. 
+
+Then, the actual situation is that before install Koordinator, a large number of clusters have been running for a period of time, and have its own Priority Class definition and supporting operating system.
+In order to be better compatible with usage habits and at the same time obtain the scheduling capability of Koordinator bound to PriorityClass, we need a compatible solution to support this situation.
+
+### Goals
+
+1. Provides a solution using Koordinator PriorityClass for already running a large number of Pod clusters.
+
+### Non-Goals/Future Work
+
+1. Provide an alternative API for PriorityClass.
+
+## Proposal
+
+### User Stories
+
+The user has already run a large number of Pods, and the PriorityClass conflicts with the Koordinator declaration.
+For example, the PriorityClass is declared as follows:
+
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: user-defined-priority-class
+value: 5000
+description: "This priority class should be used for prod service pods only."
+```
+
+Then create many Pods like this:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-name
+  labels:
+    env: test
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    imagePullPolicy: IfNotPresent
+  priorityClassName: user-defined-priority-class
+```
+
+When users install Koordinator, they cannot adjust all these Pods at once (affecting business stability), and hope that
+there is a way to regulate these Pods into the best practice category of Koordinator.
+
+### Implementation
+
+1. Defines a new label `koordinator.sh/priority-class` to compatible with Pods that are already running.
+2. When a user install Koordinator, find a way to label the existing Pods with this label. There is a default value inference mechanism for unlabeled Koordinator, but please do not rely on it.
```

Comment:
Does the new label affect the priority comparison between pods? If it does, we may need to define the priority value for each priority label.

e.g. When a pod `p0` with the new label (`koordinator.sh/priority-class=Prod`) is compared with a pod `p1` with koord priority class specified (`pod.spec.priorityClassName=koord-prod`), we need to determine which one is considered high-priority.

Reply:
It does not affect the priority comparison. There is a section in it that explains that when the priority is reversed by introducing a compatible label, you need to be aware of this problem. It is recommended to turn off priority preemption, and consider turning it on after completing the transition of the plan.